OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java

```diff
@@ -1097,12 +1108,28 @@ public boolean isCancelled() {
 
 				BinaryMapIndexReader.SearchRequest<Amenity> pointRequest = BinaryMapIndexReader.buildSearchPoiRequest(
 						0, 0, Algorithms.emptyIfNull(article.title), 0, Integer.MAX_VALUE, 0, Integer.MAX_VALUE,
-						getSearchFilter(article.getPointFilterString()),
+						getSearchFilter(article.getMainFilterString(), article.getPointFilterString()),
 						new ResultMatcher<Amenity>() {
 							@Override
 							public boolean publish(Amenity amenity) {
 								if (amenity.getRouteId().equals(article.getRouteId())) {
-									if (article.getPointFilterString().equals(ROUTE_TRACK_POINT)) {
+									if (ROUTE_TRACK.equals(amenity.getSubType())) {
+										for (String key : amenity.getAdditionalInfoKeys()) {
+											if (key.startsWith(GPX_EXTENSION_TAG_PREFIX)) {
+												String tag = key.replaceFirst(GPX_EXTENSION_TAG_PREFIX, "");
+												String val = amenity.getAdditionalInfo(key);
+												gpxFileExtensions.put(tag, val);
+											} else if (key.startsWith(POINTS_GROUPS_TAG_PREFIX)) {
+												final String DELIMITER = "~~~";
+												String joinedValues = amenity.getAdditionalInfo(key);
+												List<String> values = Arrays.asList(joinedValues.split(DELIMITER));
+												if (key.endsWith("names")) pgNames.addAll(values);
+												else if (key.endsWith("icons")) pgIcons.addAll(values);
```

Comment:
points_groups_names must be in constant in Kotlin lib

Reply:
done